Quaver.API/Maps/Processors/Difficulty/Rulesets/Keys/DifficultyProcessorKeys.cs

```diff
@@ -584,7 +585,16 @@ private float GetCoefficientValue(float duration, float xMin, float xMax, float
 
             // calculate ratio between min and max value
             var ratio = Math.Max(0, (duration - xMin) / (xMax - xMin));
-                ratio = 1 - Math.Min(1, ratio);
+            //if ratio is too big scale based on nps instead
+            if (ratio > 1)
+            {
+                if (AverageNoteDensity < 1)
+                    return .4f;
+                if (AverageNoteDensity > 4)
+                    return AverageNoteDensity * .5f;
+                return AverageNoteDensity * .266f + .134f;
+            }
```

Comment:
What do these numbers represent? Put the magic numbers into their own variables with names that describe what they are.